### PR TITLE
moving developer-only strings to api messages

### LINKED ISSUE #8574: Migrate API errors to apiMessages from i18n (no translations for developer-only messages)
https://github.com/HabitRPG/habitica/commit/939712ad1f8e8c8e228907b30844fcd3bb5dae51 introduced the `apiMessages` library  (https://github.com/HabitRPG/habitica/blob/develop/website/server/libs/apiMessages.js) because we've decided that API messages aimed at developers will only be in English from now on.

We should migrate old translated messages to the new lib.

### PR DESCRIPTION
**Pull request by negue**
closes #8574

### CODE REVIEW

**Comment by negue** [comment]
any other messages that should be moved to `apiMessages` ? @paglias  @Alys @TheHollidayInn

**Comment by paglias** [comment]
that's great! can you look into the tests failures?

**Comment by negue** [comment]
I can't figure out the reason which cause 

```
AssertionError: expected { Object (code, error, ...) } to deeply equal { Object (code, error, ...) }
      + expected - actual
       {
      -  "code": 500
      -  "error": "InternalServerError"
      -  "message": "An unexpected error occurred."
      +  "code": 400
      +  "error": "BadRequest"
      +  "message": "Invalid request parameters."
       }
```

I mean the InternalServerError doesn't log any real information what exactly seems to be broken -.- , the apiMessages call works fine and returns the string (did a console.log before), so did the express-validator "broke" somehow? 😅

**Comment by paglias** [comment]
You can run the server in one terminal with `NODE_ENV=test npm start` and in another the tests with `npm run test:api-v3:integration:separate-server`, the internal error will appear in the server terminal with all the details

File: website/common/script/ops/buy/buy.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ import hourglassPurchase from './hourglassPurchase';
 
 module.exports = function buy (user, req = {}, analytics) {
   let key = get(req, 'params.key');
-  if (!key) throw new BadRequest(i18n.t('missingKeyParam', req.language));
+  if (!key) throw new BadRequest(apiMessages('missingKeyParam', req.language));
```
Comment (paglias) [inline website/common/script/ops/buy/buy.js]: why req.language is passed here?

File: website/common/script/ops/buy/buyMysterySet.js [hunk 2]
```diff
@@ -7,10 +7,11 @@ import {
   NotAuthorized,
   NotFound,
 } from '../../libs/errors';
+import apiMessages from '../../../../server/libs/apiMessages';
 
 module.exports = function buyMysterySet (user, req = {}, analytics) {
   let key = get(req, 'params.key');
-  if (!key) throw new BadRequest(i18n.t('missingKeyParam', req.language));
+  if (!key) throw new BadRequest(apiMessages('missingKeyParam', req.language));
```
Comment (paglias) [inline website/common/script/ops/buy/buyMysterySet.js]: here too, req.language. and in other places too

**Comment by negue** [comment]
@paglias  seems I renamed two keys in order to rename all occurences but kinda forgot about it ... 🤦‍♂️ 

fixed it now

File: website/common/script/ops/buy/buyMysterySet.js (hunk 2 above)
Comment (negue) [reply website/common/script/ops/buy/buyMysterySet.js]: good point, I thought I removed them all, since those aren't needed, going to remove now

**Comment by negue** [comment]
I oversaw some `get(req, 'param.something')` calls, gonna change those too

**Comment by paglias** [comment]
@negue is this ready now?

**Comment by negue** [comment]
@paglias if the travis CI is ok then its ready

**Comment by negue** [comment]
the current test fail is fixed with #10251

**Comment by paglias** [comment]
thanks for this!

**Pull request merged**